### ISSUE TITLE
Mysql2::Error: Column 'email' cannot be null

### ISSUE THREAD

**Issue by JonathonReinhart** [SUSE/Portus#620]
**config-local.yml:**
```
ldap:
  enabled: true
  # ...
  guess_email:
    enabled: true
    attr: 'mail'
```

It turns out some of my users have this attribute in LDAP, and some do not. When a user tries to login without this attribute being set in his account,  he gets a blank page (500 error). In `production.log` I see:

```
 INFO -- : guess_email:
ERROR -- : Mysql2:Error: Column 'email' cannot be null: INSERT INTO 'users' ('username', 'email', 'encrypted_password', 'ldap_name', 'created_at', 'updated_at') VALUES ('[username]', NULL, '[hash]', '[username]', '[date]', '[date]')
```

I made this addition in `lib/portus/ldap.rb`:

```
Rails.logger.info("guess email: #{guess_email}")

user = User.create(
  username:  name,
  email:    guess_email,
    ....
```

As you can see `guess_email` is either NULL or empty; I'm not sure which.

I'm not sure the best way to handle this. I would prefer that the user *not* be given the option to enter his email address if my config indicates that it must come from LDAP.

**Comment by mssola** [comment]
IIRC, in another bug report you said that you are using Portus 1.1 from the RPM right ? If this is the case, then this issue has been fixed in the commit e49a1016db6644df8d74d2a3285b9af3ad10b6db. This is fixed in Portus 2.0.0, and as @flavio said in another thread, we will release the RPM asap :) If this was not the case, please, feel free to re-open this issue and provide which setup you're using :P

**Comment by JonathonReinhart** [comment]
@mssola You are correct. I searched pretty hard and couldn't find any references to this problem. Sorry for the noise.
